src/routes/(app)/order/[id]/+page.server.ts

```diff
@@ -97,5 +101,67 @@ export const actions = {
 		);
 
 		throw redirect(303, request.headers.get('referer') || '/');
+	},
+	addPayment: async ({ params, request, locals }) => {
+		const order = await collections.orders.findOne({
+			_id: params.id
+		});
+
+		if (!order) {
+			throw error(404, 'Order not found');
+		}
+
+		if (order.status !== 'pending') {
+			throw error(400, 'Order is not pending');
+		}
+
+		const remainingAmount = orderAmountWithNoPayments(order);
+
+		if (remainingAmount <= 0) {
+			throw error(400, 'Order has no remaining amount to pay');
+		}
+
+		let methods = paymentMethods({ role: locals.user?.roleId });
+
+		for (const item of order.items) {
+			if (item.product.paymentMethods) {
+				methods = methods.filter((method) => item.product.paymentMethods?.includes(method));
+			}
+		}
+
+		if (!methods.length) {
+			throw error(400, 'No payment methods available');
+		}
+
+		const formData = await request.formData();
+		const parsed = z
+			.object({
+				amount: z.string().regex(/^\d+(\.\d+)?$/),
+				method: z.enum(methods as [PaymentMethod, ...PaymentMethod[]])
+			})
+			.parse({
+				amount: formData.get('amount'),
+				method: formData.get('method')
+			});
+
+		const amount = parsePriceAmount(parsed.amount, order.currencySnapshot.main.totalPrice.currency);
+		if (amount <= 0) {
+			throw error(400, 'Invalid amount');
+		}
+		if (amount > remainingAmount) {
+			throw error(400, 'Amount is greater than the remaining amount to pay');
+		}
+
+		await addOrderPayment(
+			order,
+			parsed.method,
+			{
+				amount,
+				currency: order.currencySnapshot.main.totalPrice.currency
+			},
+			{ expiresAt: null }
```

Comment:
`expiresAt` should be similar to here: https://github.com/be-BOP-io-SA/be-BOP/blob/66eee4d68ed507b8f5a13ff13f62256191493fb2/src/lib/server/orders.ts#L996-L998 ?

(instead of `null`)